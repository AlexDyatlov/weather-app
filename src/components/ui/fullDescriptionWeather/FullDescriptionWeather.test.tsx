import { render } from '@testing-library/react';
import FullDescriptionWeather from './fullDescriptionWeather';

import { IWeekWeather } from '../../../@types/weekWeather.interface';

const mockFullDescriptionWeatherData: IWeekWeather = {
  dt: '1722456000',
  clouds: 58,
  weather: [{ description: 'облачно с прояснениями', icon: '04d' }],
  humidity: 79,
  wind_speed: 2.69,
  temp: {
    day: 16.06,
    min: 14.3,
    max: 23.01,
    night: 15.22,
    eve: 21.01,
    morn: 14.4
  },
  feels_like: { min: 21, max: 10, day: 16.1, eve: 20.94, morn: 14.45, night: 15.2 }
};

describe('FullDescriptionWeather component', () => {
  it('render', () => {
    const { container } = render(<FullDescriptionWeather {...mockFullDescriptionWeatherData} />);
    expect(container).toMatchSnapshot();
  });
});
