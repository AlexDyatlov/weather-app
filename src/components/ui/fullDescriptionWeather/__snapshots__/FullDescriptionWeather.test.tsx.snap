// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullDescriptionWeather component render 1`] = `
<div>
  <p>
    Облачность 
    58
    %, 
    облачно с прояснениями
     
  </p>
  <p>
    Влажность 
    79
     %
  </p>
  <p>
    Скорость ветра 
    2.7
     метр/сек
  </p>
  <div
    class="pb-4"
  >
    <ul
      class="flex gap-3 border-b border-b-slate-400 pb-2 text-center"
    >
      <li
        class="w-full max-w-[180px]"
      />
      <li
        class="w-full max-w-[80px]"
      >
        Утро
      </li>
      <li
        class="w-full max-w-[80px]"
      >
        День
      </li>
      <li
        class="w-full max-w-[80px]"
      >
        Вечер
      </li>
      <li
        class="w-full max-w-[80px]"
      >
        Ночь
      </li>
    </ul>
    <ul>
      <li
        class="flex gap-3 py-2 text-center"
      >
        <div
          class="w-full max-w-[180px] text-left"
        >
          Температура:
        </div>
        <div
          class="w-full max-w-[80px]"
        >
          14
          °C
        </div>
        <div
          class="w-full max-w-[80px]"
        >
          16
          °C
        </div>
        <div
          class="w-full max-w-[80px]"
        >
          21
          °C
        </div>
        <div
          class="w-full max-w-[80px]"
        >
          15
          °C
        </div>
      </li>
      <li
        class="flex gap-3 text-center"
      >
        <div
          class="w-full max-w-[180px] text-left"
        >
          Ощущается как:
        </div>
        <div
          class="w-full max-w-[80px]"
        >
          14
          °C
        </div>
        <div
          class="w-full max-w-[80px]"
        >
          16
          °C
        </div>
        <div
          class="w-full max-w-[80px]"
        >
          21
          °C
        </div>
        <div
          class="w-full max-w-[80px]"
        >
          15
          °C
        </div>
      </li>
    </ul>
  </div>
</div>
`;
